### PR TITLE
feat(presets): internalize presets

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

**Comment by rarkins** [comment]
- [ ] Needs validation added

File: lib/config/presets/internal/index.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import * as defaultPreset from './default';
+import * as dockerPreset from './docker';
+import * as groupPreset from './group';
+import * as helpersPreset from './helpers';
+import * as monorepoPreset from './monorepo';
+import * as packagesPreset from './packages';
+import * as previewPreset from './preview';
+import * as schedulePreset from './schedule';
+
+/* eslint-disable global-require */
```
Comment (viceice) [inline lib/config/presets/internal/index.ts:12]:
```suggestion
```
Here is no require.

File: lib/config/presets/internal/index.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  docker: dockerPreset.presets,
+  group: groupPreset.presets,
+  helpers: helpersPreset.presets,
+  monorepo: monorepoPreset.presets,
+  packages: packagesPreset.presets,
+  preview: previewPreset.presets,
+  schedule: schedulePreset.presets,
+};
+
+export function getPreset(pkgName: string, presetName: string): Preset {
```
Comment (viceice) [inline lib/config/presets/internal/index.ts:25]:
```suggestion
export function getPreset(pkgName: string, presetName: string): Preset | undefined {
```

File: lib/config/presets/internal/index.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import * as dockerPreset from './docker';
+import * as groupPreset from './group';
+import * as helpersPreset from './helpers';
+import * as monorepoPreset from './monorepo';
+import * as packagesPreset from './packages';
+import * as previewPreset from './preview';
+import * as schedulePreset from './schedule';
+
+/* eslint-disable global-require */
+const groups: Record<string, Record<string, any>> = {
```
Comment (viceice) [inline lib/config/presets/internal/index.ts:13]: We should optimize types later

File: lib/config/presets/internal/config.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+export const presets = {
```
Comment (viceice) [inline lib/config/presets/internal/config.ts:1]:
```suggestion
import { Preset } from '../common';
export const presets : Record<string, Preset> = {
```

or fixing later ?

File: lib/config/presets/internal/index.spec.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import * as internal from '.';
+import { validateConfig } from '../../validation';
+import { massageConfig } from '../../massage';
+
+jest.mock('../../../datasource/npm');
+
+const ignoredPresets = ['default:group', 'default:timezone'];
```
Comment (viceice) [inline lib/config/presets/internal/index.spec.ts:7]: What are the problems with this presets?

**Review by viceice — APPROVED** [review]

File: lib/config/presets/internal/index.spec.ts (hunk 5 above)
Comment (rarkins) [reply lib/config/presets/internal/index.spec.ts:7]: They use `{{arg0}}` in their values and fail validation. `default:group` tries to look up a non-existent preset while `default:timezone` is validated against valid timezones.

**Pull request merged**